src/time.rs

```diff
@@ -0,0 +1,242 @@
+use crate::sys::time::TimeSpec;
+#[cfg(target_os = "linux")]
+use crate::unistd::Pid;
+#[cfg(target_os = "linux")]
+use crate::Error;
+use crate::{Errno, Result};
+use libc::{self, clockid_t};
+use std::mem::MaybeUninit;
+
+/// Clock identifier
+///
+/// Newtype pattern around `clockid_t` (which is just alias). It pervents bugs caused by
+/// accidentally passing wrong value.
+#[derive(Debug, Copy, Clone, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub struct ClockId(clockid_t);
+
+impl ClockId {
+    /// Creates `ClockId` from raw `clockid_t`
+    pub fn from_raw(clk_id: clockid_t) -> Self {
+        ClockId(clk_id)
+    }
+
+    /// Returns `ClockId` of a `pid` CPU-time clock
+    #[cfg(target_os = "linux")]
+    pub fn pid_cpu_clock_id(pid: Pid) -> Result<Self> {
+        clock_getcpuclockid(pid)
```

Comment:
This function is supported by FreeBSD, too.

Reply:
Indeed, although it's not in rust-lang/libc bindings for FreeBSD. I'll try to add it and update the PR.

Reply:
[Added to more targets](https://github.com/rust-lang/libc/pull/1867) in libc waiting for a [version bump](https://github.com/rust-lang/libc/pull/1879) to update this PR.